.gitignore

```diff
@@ -1,3 +1,3 @@
 node_modules
 npm-debug.log
-
+coverage
```

Comment:
I like to maintain single ignore file too. 
But, as per [this](https://yarnpkg.com/en/docs/yarn-lock#toc-check-into-source-control) `yarn.lock` should be checked in. This ensures that anyone running yarn will get exact set of transitive dependencies (like npm shrinkwrap). 
`.vscode` is ide configuration for debugging, so I thought it would make it handy to quickly debug test/source. 
but neither of these needs to be published on NPM, so you will still have to maintain `.npmignore`